### PR TITLE
Static Utils & unique file generation for Xray

### PR DESCRIPTION
**Pull request by hmitgang**
- Utils methods refactored to be static. Calls to these methods refactored to static calls.
- Xray data generation creates unique artifacts (with unique sha256s). These are uploaded to Artifactory but deleted locally at the end of the test suite.

### CODE REVIEW

**Review by MahithaB — APPROVED** [review]

File: src/test/groovy/tests/GenerateXrayDataTest.groovy [hunk 1]
```diff
@@ -82,8 +86,41 @@ class GenerateXrayDataTest extends XraySteps{
         Reporter.log("- Create repositories for HA distribution. Successfully created", true)
     }
 
-    @Test(priority=2, groups=["xray_generate_data"], testName = "Create policy and watch. Assign policy to watch")
-    void createPolicyTest(){
+    @Test(priority=2, groups=["xray_generate_data"], testName = "Delete existing test policies and watches")
+    void deleteExistingWatches() {
+        def watches = given()
```
Comment (danielmkn) [inline src/test/groovy/tests/GenerateXrayDataTest.groovy:91]: please move it to the steps class

File: src/test/groovy/tests/GenerateXrayDataTest.groovy [hunk 2]
```diff
@@ -82,8 +86,41 @@ class GenerateXrayDataTest extends XraySteps{
...
+                .auth().preemptive().basic(username, password)
+                .when().get("${xrayBaseUrl}/v2/policies")
+                .then().extract().body().jsonPath().getList("\$").stream()
+                .map({it.getAt("name").toString()})
+                .filter({it.startsWith(namePrefix)})
+                .collect()
+        policies.forEach { name ->
+            given().auth().preemptive().basic(username, password)
+                .when().delete("${xrayBaseUrl}/v2/policies/${name}")
+                .then().log().ifValidationFails().assertThat().statusCode(200)
```
Comment (danielmkn) [inline src/test/groovy/tests/GenerateXrayDataTest.groovy:116]: assertThat is redundant here

File: src/test/groovy/tests/GenerateXrayDataTest.groovy [hunk 3]
```diff
@@ -82,8 +86,41 @@ class GenerateXrayDataTest extends XraySteps{
...
+                .auth().preemptive().basic(username, password)
+                .when().get("${xrayBaseUrl}/v2/watches")
+                .then().extract().body().jsonPath().getList("\$").stream()
+                .map({it.getAt("general_data").getAt("name").toString()})
+                .filter({it.startsWith(namePrefix)})
+                .collect()
+        watches.forEach { name ->
+            given().auth().preemptive().basic(username, password)
+                .when().delete("${xrayBaseUrl}/v2/watches/${name}")
+                .then().assertThat().statusCode(200)
```
Comment (danielmkn) [inline src/test/groovy/tests/GenerateXrayDataTest.groovy:101]: assertThat is redundant

File: src/test/groovy/tests/GenerateXrayDataTest.groovy [hunk 4]
```diff
@@ -155,27 +192,38 @@ class GenerateXrayDataTest extends XraySteps{
...
+        for (i in artifactIDs) {
             def artifactName = artifactFormat(i)
             def artifact = new File("${artifactsPath}${artifactName}")
             def sha256 = Utils.generateSHA256(artifact)
-            sleep(2000)  // UI requests are finicky. Let server settle.
-            Response response = xraySteps.assignLicenseToArtifact(UILoginHeaders, artifactoryBaseURL, artifactName, sha256, license_name, liense_full_name, license_references)
-            response.then().log().ifValidationFails().statusCode(200)
+            sleep(1000)  // UI requests are finicky. Let server settle.
+            int tries = 0
+            Awaitility.await().atMost(120, TimeUnit.SECONDS).with()
```
Comment (danielmkn) [inline src/test/groovy/tests/GenerateXrayDataTest.groovy:208]: this should be in the steps class

**Review by danielmkn — CHANGES_REQUESTED** [review]

**Review by danielmkn — APPROVED** [review]

**Review by MahithaB — APPROVED** [review]

**Pull request merged**